src/ImageSharp/Formats/Webp/Lossless/LosslessUtils.cs

```diff
@@ -759,28 +760,184 @@ public static void BundleColorMap(Span<byte> row, int width, int xBits, Span<uin
         /// <returns>Shanon entropy.</returns>
         public static float CombinedShannonEntropy(Span<int> x, Span<int> y)
         {
-            double retVal = 0.0d;
-            uint sumX = 0, sumXY = 0;
-            for (int i = 0; i < 256; i++)
+#if SUPPORTS_RUNTIME_INTRINSICS
+            if (Avx2.IsSupported)
             {
-                uint xi = (uint)x[i];
-                if (xi != 0)
+                double retVal = 0.0d;
+                Span<int> tmp = stackalloc int[8];
+                ref int xRef = ref MemoryMarshal.GetReference(x);
+                ref int yRef = ref MemoryMarshal.GetReference(y);
+                Vector256<int> sumXY256 = Vector256<int>.Zero;
+                Vector256<int> sumX256 = Vector256<int>.Zero;
+                ref int tmpRef = ref MemoryMarshal.GetReference(tmp);
```

Comment:
`Span<int> tmp` is only used to get `ref int tmpRef`, so we could just avoid the span at all, including the stack-cookie check the JIT needs to emit (to detect stack overruns).

```suggestion
                Vector256<int> tmp = Vector256<int>.Zero;    // has the size of the scratch space of sizeof(int) * 8
                ref int xRef = ref MemoryMarshal.GetReference(x);
                ref int yRef = ref MemoryMarshal.GetReference(y);
                Vector256<int> sumXY256 = Vector256<int>.Zero;
                Vector256<int> sumX256 = Vector256<int>.Zero;
                ref int tmpRef = ref Unsafe.As<Vector256<int>, int>(ref tmp);
```

For comparison see [sharplap](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCUBXAOwwEt8YLAMIR8AB14AbGFADKMgG68wMXAG4a9Jq049+ggJI8ZEMfKhKV6zY2btufASyMYovLrmXXqmgMxNSBiEGAG8aBgimf3cMBgBZOgAKAEoGcMiw6kjshlkxbC4AHhiAPgYMcQYAXgZcDGwwAGtsSUkIMAYYgG0ADgBdDSyciNgAM06ecvE2GHGasfiYfGgATzjsKFwACxaWAHEYDBnRmRguFUSKsWTB4Yj04eIAdgYAVQ9sE5YAQQATX8SCyuxzQDHINweDAAvpDIV1ZI1eGIADLtFq4Iy8DB9SHEaKTOLkFJpIYRTJ3XL5IqlKZiaq1epNFptDrdfq3O4LGK0470hZxJardabHaSfaHY6nc4wS7iCGknKQ7LPN4fL5/AFA6azUHgjkRGEK3H42JxUgpSHku4ANRgYAw0FIAFYkMUeGVfhx8PgVvTfrNsBxJBh9Tk8gU3Rgyld6QLllA1httrshLBsBgYOGuIDZqrcJ9BN9cIVbfbHS7IyVQaUc+NPd6VslQT0bhSlZEuZNgbn5rm40Kk6LxUdZlKLld5W2FcqXu98+r/rWeTqwZPIobssaJqbfBaFVbhqWHVBna6afWfX6A0GQ+2Rrnud25gwFnOCz9i0fy2f3dX3UuL0bUNN2nSIVTfBdNVzJ9dTXA0aChIA=) (method _M3_ is used here).

Reply:
that's a neat trick